test/unit/org/apache/cassandra/locator/DynamicEndpointSnitchTest.java

```diff
@@ -18,35 +18,99 @@
 
 package org.apache.cassandra.locator;
 
-import java.io.IOException;
+import java.net.UnknownHostException;
```

Comment:
It looks like shutting down the DES is covered a little in the settings test, but it would be good if it validated somewhere the futures are cleaned up and there are no scheduled/running tasks in the executors.

Reply:
My code coverage is a bit wonky today, but is sendPingMessage tested? Is it possible?

Reply:
> but it would be good if it validated somewhere the futures are cleaned up and there are no scheduled/running tasks in the executors.

ack, writing that

> but is sendPingMessage tested? Is it possible?
It's not tested because it'll require the MessagingService, but I can just setup some sinks for testing. Working on it.

Reply:
Ok, I've added tests by using a message sink, but the test code is a bit gnarly (and very "wait until this thread somewhere does a thing"). As currently implemented we do a true "integration" test to make sure that it works e2e but I don't like it very much. If you're ok with it I'd rather just make the latency probe sending method take some functions and then I can just unit test it (like instead of calling messaging service methods directly it would be passed them). Do you prefer the e2e approach or the unit approach?